ds/beis_indicators/defra/make_defra_indicators.py

```diff
@@ -0,0 +1,153 @@
+    
+    if not os.path.isdir(nuts_dir):
+        with ZipFile(f'{nuts_dir}.zip','r') as archive:
+            archive.extractall(nuts_dir)
+        
+    nuts_fin = (f'{nuts_dir}/'
+                f'NUTS_RG_{resolution}M_{year}_{projection}_LEVL_{level}.shp')
+    nuts_gdf = gpd.read_file(nuts_fin)
+    
+    if countries is not None:
+        nuts_gdf = nuts_gdf.set_index('CNTR_CODE').loc[countries].reset_index()
+        
+    return nuts_gdf
+
+def make_air_pollution_nuts(years, pollution_type='pm10', nuts_level=2, aggfunc=np.mean):
+    '''make_air_pollution_nuts
+    Creates air pollution indicator for NUTS regions over a range of years, using
+    modelled point data from DEFRA at a 1km resolution.
+    
+    Output is a DataFrame of aggregated values at the level of NUTS regions.
+    
+    Args:
+        years (iter of int): Collects pollution data and creates indicators over
+            this range of years.
+        pollution_type (str): Name of pollutant. Choices are currently pm10 
+            and pm25, no2 or nox. Defaults to pm10.
+        nuts_level (int): NUTS region level. Can be 1, 2 or 3.
+        aggfunc (function): Function used to aggregate point data within a region, 
+            for example finding the average, maximum or percentile value. Default 
+            is np.mean.
+            
+    Returns:
+        df (pandas.DataFrame): DataFrame of processed indicator.
+    '''
+    
+    pin = 'epsg:27700'
+    pout = 'epsg:4326'
+    
+    out_dir = f'{data_path}/processed/defra/'
+    raw_data_dir = f'{project_dir}/data/raw/defra/'
+    
+    dfs = []
+    for year in years:
+        logger.info(f'Creating {year} {pollution_type} indicator for NUTS{nuts_level}')
+        pollution = load_pollution_data(year, pollution_type)
+        pollution['longitude'], pollution['latitude'] = translate_coordinates(
+            pollution['x'].values, pollution['y'].values, pin, pout)
+        pollution_gdf = coordinates_to_points(pollution, 'latitude', 'longitude')
+        
+        nuts_spec_year = nuts_earliest(year)
+        nuts = load_nuts_regions(year=nuts_spec_year, level=nuts_level)
+        nuts = nuts.to_crs(pout.upper())
+        
+        pollution_gdf = gpd.sjoin(pollution_gdf, nuts, op='within')
+        
+        aggregated = (pollution_gdf.groupby('NUTS_ID')[pollution_type]
+                      .apply(aggfunc)
+                      .reset_index())
+        aggregated['nuts_year_spec'] = nuts_spec_year
+        aggregated['year'] = year
+        
+        value_header = f'air_pollution_{aggfunc.__name__}_{pollution_type}'
+        headers = {
+            pollution_type: value_header,
+            'NUTS_ID': 'nuts_id'
+        }
+        aggregated = aggregated.rename(columns=headers)
+        dfs.append(aggregated)
+    
+    df = pd.concat(dfs)
+    df = df[['year', 'nuts_id', 'nuts_year_spec', value_header]]
+    df.to_csv(f'{out_dir}{value_header}.nuts{nuts_level}.csv', index=False)
```

Comment:
it would be nice if this function returned `df` and another one saved it (see `save_indicator`, which makes me realise you need to rebase this branch to `dev` as it's still based off from 28ee6de)